contracts/contracts/strategies/plume/RoosterAMOStrategy.sol

```diff
@@ -0,0 +1,1181 @@
+        IERC20(WETH).safeTransfer(_recipient, _amount);
+        emit Withdrawal(WETH, address(0), _amount);
+    }
+
+    /**
+     * @dev Retuns bool indicating whether asset is supported by strategy
+     * @param _asset Address of the asset
+     */
+    function supportsAsset(address _asset) public view override returns (bool) {
+        return _asset == WETH;
+    }
+
+    /**
+     * @dev Approve the spending amounts for the assets
+     */
+    function _approveTokenAmounts(
+        uint256 _wethAllowance,
+        uint256 _oethBAllowance
+    ) internal {
+        IERC20(WETH).approve(address(liquidityManager), _wethAllowance);
+        IERC20(OETHp).approve(address(liquidityManager), _oethBAllowance);
+    }
+
+    /***************************************
+              Liquidity management
+    ****************************************/
+    /**
+     * @dev Add liquidity into the pool in the pre-configured WETH to OETHp share ratios
+     * defined by the allowedPoolWethShareStart|End interval.
+     *
+     * Normally a PoolLens contract is used to prepare the parameters to add liquidity to the
+     * Rooster pools. It has some errors when doing those calculation and for that reason a
+     * much more accurate Quoter contract is used. This is possible due to our requirement of
+     * adding liquidity only to one tick - PoolLens supports adding liquidity into multiple ticks
+     * using different distribution ratios.
+     */
+    function _addLiquidity() internal {
+        uint256 _wethBalance = IERC20(WETH).balanceOf(address(this));
+        uint256 _oethBalance = IERC20(OETHp).balanceOf(address(this));
+        // don't deposit small liquidity amounts
+        if (_wethBalance <= MIN_BALANCE_THRESHOLD) {
+            return;
+        }
+
+        (
+            bytes memory packedSqrtPriceBreaks,
+            bytes[] memory packedArgs,
+            ,
+            uint256 OETHpRequired
+        ) = _getAddLiquidityParams(_wethBalance, 1e30);
+
+        if (OETHpRequired > _oethBalance) {
+            IVault(vaultAddress).mintForStrategy(OETHpRequired - _oethBalance);
+        }
+
+        _approveTokenAmounts(_wethBalance, OETHpRequired);
+
+        (
+            uint256 _wethAmount,
+            uint256 _oethAmount,
+            uint32[] memory binIds
+        ) = liquidityManager.addPositionLiquidityToSenderByTokenIndex(
+                mPool,
+                0, // NFT token index
+                packedSqrtPriceBreaks,
+                packedArgs
+            );
+
+        require(binIds.length == 1, "Unexpected binIds length");
+
+        emit LiquidityAdded(
+            _wethBalance, // wethAmountDesired
+            OETHpRequired, // oethpAmountDesired
+            _wethAmount, // wethAmountSupplied
+            _oethAmount, // oethbAmountSupplied
+            tokenId, // tokenId
+            underlyingAssets
+        );
+
+        // burn remaining OETHp and skip check because liquidityManager takes
```

Comment:
What check is being skipped here?

Reply:
`BurnOethOnTheContract` used to take a parameter to skip the minimal threshold check. That has been removed some time ago. Updated the comment, thanks: https://github.com/OriginProtocol/origin-dollar/pull/2499/commits/51cd18bdf9ad23e66fbb500e405e70174dfc6ecd